### ISSUE TITLE
Error when message length > 8192 (R 3.4.1)

### ISSUE THREAD

**Issue by kishore5790** [WLOGSolutions/r-logging#2]
Steps to reproduce:

`msg <- "a"; for(i in 1:13) msg <- paste0(msg,msg)`
`msg <- paste0(msg, "a") ## length 8193`
`logging::loginfo(msg) ## sprintf crashes - core dump`

R --version output:

R version 3.4.1 (2017-06-30) -- "Single Candle"
Copyright (C) 2017 The R Foundation for Statistical Computing
Platform: x86_64-redhat-linux-gnu (64-bit)
R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under the terms of the
GNU General Public License versions 2 or 3.
For more information about these matters see
http://www.gnu.org/licenses/.

**Comment by ws171913** [comment]
Hello,

sprintf has limitation to 8192 chars on fmt argument on Linux distributions

On ubuntu14.04 with GNU R the code you provided does not crush R just raises regular exception. But as I can deduce you are using Revolution R. I've tried R v3.4.1 (Single Candle) on Ubuntu 14.04. Unfortunatly I could not retproduce crush: it also gracefully raises exception "'fmt' length exceeds maximal format length 8192"

Regarding your proposed solution. It fixes basic problem: just logging message without formatting. But the problem still occures if any formating is enforced with long fmt. Please try this:

msg <- paste0(c(rep("a", 8192), "%s"), collapse = "") ## length 8192 + %s
logging::loginfo(msg, "Ok") # fails with fmt' length exceeds maximal format length 8192

You fix handles logging without formating. But it does not raise 'too few arguments' exception if formatting arguments are present in fmt and not provided (as sprintf regulary does). It is important information for developer. Please try this:
logging::loginfo(msg) # should raise 'too few arguments'

To summarize: you fix is workaround in fact. I need some time to find complete solution and update tests.  I'll get bat to you till the end of week.

**Comment by jamesdalg** [comment]
I just wanted to say that my package is now experiencing this particular issue as well. I'll see if I can't come up with a workaround, but you may know a way better than I. https://github.com/jamesdalg/CNVScope/issues/1
